src/NHibernate.Test/SystemTransactions/ResourceManagerFixture.cs

```diff
@@ -0,0 +1,814 @@
+				// for delegating the resource management to the SQL server.)
+				EnlistResource.EnlistDurable(false, true);
+				_log.InfoFormat(
+					"Fake connection opened, scope id {0} and distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				// Simulate another resource, not even supporting single phase
+				EnlistResource.EnlistDurable();
+				_log.InfoFormat(
+					"Fake other resource, scope id {0} and distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				// Simulate NHibernate : volatile no single phase support
+				if (enlistInPrepare)
+					EnlistResource.EnlistWithPrepareEnlistmentVolatile();
+				else
+					EnlistResource.EnlistVolatile();
+				_log.InfoFormat(
+					"Fake session opened, scope id {0} and distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				scope.Complete();
+				_log.Info("Scope completed");
+			}
+			_log.Info("Scope disposed");
+		}
+		
+		[Test]
+		public void DistributedNpgsqlCommit([Values(false, true)] bool enlistInPrepare)
+		{
+			using (var scope = new TransactionScope())
+			{
+				_log.InfoFormat(
+					"Scope opened, id {0}, distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				// Simulate a Npgsql connection: as of Npgsql 3.2.4, volatile resource with single phase support
+				EnlistResource.EnlistVolatile(false, true);
+				_log.InfoFormat(
+					"Fake connection opened, scope id {0} and distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				// Simulate another resource, not even supporting single phase (required for going distributed with "Npgsql")
+				EnlistResource.EnlistDurable();
+				_log.InfoFormat(
+					"Fake other resource, scope id {0} and distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				// Simulate NHibernate : volatile no single phase support
+				if (enlistInPrepare)
+					EnlistResource.EnlistWithPrepareEnlistmentVolatile();
+				else
+					EnlistResource.EnlistVolatile();
+				_log.InfoFormat(
+					"Fake session opened, scope id {0} and distributed id {1}",
+					SysTran.Current.TransactionInformation.LocalIdentifier,
+					SysTran.Current.TransactionInformation.DistributedIdentifier);
+				scope.Complete();
+				_log.Info("Scope completed");
+			}
+			_log.Info("Scope disposed");
+		}
+
+		#endregion
+
+		#region Failure
+
+		[Test]
+		[Explicit("Failing.")]
+		public void DistributedTransactionStatusMustBeInactiveAfterRollbackedScope()
```

Comment:
Here the test case for the transaction status bug.

Good news: that occurs only with cloned transaction. So we can store yet another member for additionally checking if the original transaction is still active and disregard a cloned active status.

By the way, I highly suspect the `ObjectDisposedException` failure was occurring because `TransactionCompletion`was checking e.Transaction instead of the cloned transaction, and that this event transaction is the original on instead of the cloned one. I am going to add another test to demonstrate that later point.